bot/cogs/watchchannels/watchchannel.py

```diff
@@ -0,0 +1,353 @@
+        """
+        try:
+            data = await self.bot.api_client.get(self.api_endpoint, params=self.api_default_params)
+        except aiohttp.ClientResponseError as e:
+            self.log.exception(f"Failed to fetch the watched users from the API", exc_info=e)
+            return False
+
+        self.watched_users = defaultdict(dict)
+
+        for entry in data:
+            user_id = entry.pop('user')
+            self.watched_users[user_id] = entry
+
+        return True
+
+    async def on_message(self, msg: Message) -> None:
+        """Queues up messages sent by watched users."""
+        if msg.author.id in self.watched_users:
+            if not self.consuming_messages:
+                self._consume_task = self.bot.loop.create_task(self.consume_messages())
+
+            self.log.trace(f"Received message: {msg.content} ({len(msg.attachments)} attachments)")
+            self.message_queue[msg.author.id][msg.channel.id].append(msg)
+
+    async def consume_messages(self, delay_consumption: bool = True) -> None:
+        """Consumes the message queues to log watched users' messages."""
+        if delay_consumption:
+            self.log.trace(f"Sleeping {BigBrotherConfig.log_delay} seconds before consuming message queue")
+            await asyncio.sleep(BigBrotherConfig.log_delay)
+
+        self.log.trace(f"{self.__class__.__name__} started consuming the message queue")
+
+        # If the previous consumption Task failed, first consume the existing comsumption_queue
+        if not self.consumption_queue:
+            self.consumption_queue = self.message_queue.copy()
+            self.message_queue.clear()
+
+        for user_channel_queues in self.consumption_queue.values():
+            for channel_queue in user_channel_queues.values():
+                while channel_queue:
+                    msg = channel_queue.popleft()
+
+                    self.log.trace(f"Consuming message {msg.id} ({len(msg.attachments)} attachments)")
+                    await self.relay_message(msg)
+
+        self.consumption_queue.clear()
+
+        if self.message_queue:
+            self.log.trace("Channel queue not empty: Continuing consuming queues")
+            self._consume_task = self.bot.loop.create_task(self.consume_messages(delay_consumption=False))
+        else:
+            self.log.trace("Done consuming messages.")
+
+    async def webhook_send(
+        self,
+        content: Optional[str] = None,
+        username: Optional[str] = None,
+        avatar_url: Optional[str] = None,
+        embed: Optional[Embed] = None,
+    ) -> None:
+        """Sends a message to the webhook with the specified kwargs."""
+        try:
+            await self.webhook.send(content=content, username=username, avatar_url=avatar_url, embed=embed)
+        except discord.HTTPException as exc:
+            self.log.exception(
+                f"Failed to send message to {self.__class__.__name__} webhook",
```

Comment:
Remove the name from the message.

```suggestion
                f"Failed to send a message to the webhook",
```